File: backend/app/domain/document_templates.py
"""Plantillas de documentación clínica por tipo (etiquetas UI + códigos FHIR LOINC)."""
from __future__ import annotations

from app.domain.enums import ConsultationType

# Títulos de documento
DOCUMENT_TITLES: dict[ConsultationType, str] = {
    ConsultationType.admission_interview: "Historia clínica de ingreso",
    ConsultationType.treatment_orders: "Indicaciones médicas de tratamiento",
    ConsultationType.evolution: "Nota de evolución",
}

# LOINC para Composition.type (FHIR R4 — EU eHealth Network)
FHIR_LOINC: dict[ConsultationType, dict[str, str]] = {
    ConsultationType.admission_interview: {
        "system": "http://loinc.org",
        "code": "47039-3",
        "display": "Hospital Admission history and physical note",
    },
    ConsultationType.treatment_orders: {
        "system": "http://loinc.org",
        "code": "18776-5",
        "display": "Plan of care note",
    },
    ConsultationType.evolution: {
        "system": "http://loinc.org",
        "code": "11506-3",
        "display": "Progress note",
    },
}

# Etiquetas por campo interno del modelo tipado
SECTION_LABELS: dict[ConsultationType, dict[str, str]] = {
    ConsultationType.admission_interview: {
        "motivo_ingreso": "Motivo de ingreso",
        "enfermedad_actual": "Enfermedad actual",
        "antecedentes": "Antecedentes (personales, familiares, alergias, medicación)",
        "exploracion_fisica": "Exploración física",
        "pruebas_complementarias": "Pruebas complementarias",
        "juicio_clinico": "Juicio clínico (borrador)",
        "plan": "Plan de ingreso y actuación",
    },
    ConsultationType.treatment_orders: {
        "contexto": "Contexto del paciente",
        "indicaciones_farmacologicas": "Indicaciones farmacológicas",
        "indicaciones_no_farmacologicas": "Indicaciones no farmacológicas y cuidados",
        "vigilancia": "Vigilancia y constantes",
        "observaciones": "Observaciones y prioridad",
    },
    ConsultationType.evolution: {
        "subjetivo": "Subjetivo",
        "objetivo": "Objetivo y exploración",
        "evolucion": "Evolución clínica",
        "juicio_clinico": "Juicio clínico (borrador)",
        "plan": "Plan terapéutico y próximos pasos",
    },
}


def section_labels(consultation_type: ConsultationType) -> dict[str, str]:
    return SECTION_LABELS[consultation_type]


def document_title(consultation_type: ConsultationType) -> str:
    return DOCUMENT_TITLES[consultation_type]


def fhir_composition_type(consultation_type: ConsultationType) -> dict[str, str]:
    return FHIR_LOINC[consultation_type]

File: frontend/lib/models/document_templates.dart
import 'package:escriba_clinico/models/consultation_type.dart';

/// Definición de un campo de la planilla clínica (espejo del backend).
class DocumentSectionDef {
  const DocumentSectionDef({
    required this.key,
    required this.label,
    required this.hint,
  });

  final String key;
  final String label;
  final String hint;
}

/// Plantillas ordenadas por tipo de documento (LOINC / estándar hospitalario UE).
abstract final class DocumentTemplates {
  static const admission = [
    DocumentSectionDef(
      key: 'motivo_ingreso',
      label: 'Motivo de ingreso',
      hint: 'Motivo principal que origina el ingreso hospitalario.',
    ),
    DocumentSectionDef(
      key: 'enfermedad_actual',
      label: 'Enfermedad actual',
      hint: 'Cronología y características del cuadro actual.',
    ),
    DocumentSectionDef(
      key: 'antecedentes',
      label: 'Antecedentes',
      hint: 'Personales, familiares, alergias y medicación habitual.',
    ),
    DocumentSectionDef(
      key: 'exploracion_fisica',
      label: 'Exploración física',
      hint: 'Hallazgos objetivos de la exploración.',
    ),
    DocumentSectionDef(
      key: 'pruebas_complementarias',
      label: 'Pruebas complementarias',
      hint: 'Analítica, imagen u otras pruebas relevantes.',
    ),
    DocumentSectionDef(
      key: 'juicio_clinico',
      label: 'Juicio clínico (borrador)',
      hint: 'Impresión diagnóstica preliminar. Requiere validación del médico.',
    ),
    DocumentSectionDef(
      key: 'plan',
      label: 'Plan de ingreso y actuación',
      hint: 'Conducta terapéutica, estudios pendientes y criterios de ingreso.',
    ),
  ];

  static const treatmentOrders = [
    DocumentSectionDef(
      key: 'contexto',
      label: 'Contexto del paciente',
      hint: 'Situación clínica actual del paciente ingresado.',
    ),
    DocumentSectionDef(
      key: 'indicaciones_farmacologicas',
      label: 'Indicaciones farmacológicas',
      hint: 'Fármacos, dosis, vía y frecuencia.',
    ),
    DocumentSectionDef(
      key: 'indicaciones_no_farmacologicas',
      label: 'Indicaciones no farmacológicas y cuidados',
      hint: 'Dieta, movilización, curas y otras medidas.',
    ),
    DocumentSectionDef(
      key: 'vigilancia',
      label: 'Vigilancia y constantes',
      hint: 'Controles, monitorización y alertas.',
    ),
    DocumentSectionDef(
      key: 'observaciones',
      label: 'Observaciones y prioridad',
      hint: 'Notas adicionales o prioridad de actuación.',
    ),
  ];

  static const evolution = [
    DocumentSectionDef(
      key: 'subjetivo',
      label: 'Subjetivo',
      hint: 'Síntomas referidos por el paciente.',
    ),
    DocumentSectionDef(
      key: 'objetivo',
      label: 'Objetivo y exploración',
      hint: 'Constantes, exploración y datos objetivos.',
    ),
    DocumentSectionDef(
      key: 'evolucion',
      label: 'Evolución clínica',
      hint: 'Curso del cuadro desde el último registro.',
    ),
    DocumentSectionDef(
      key: 'juicio_clinico',
      label: 'Juicio clínico (borrador)',
      hint: 'Impresión diagnóstica actual. Requiere validación del médico.',
    ),
    DocumentSectionDef(
      key: 'plan',
      label: 'Plan terapéutico y próximos pasos',
      hint: 'Cambios de tratamiento y seguimiento.',
    ),
  ];

  static List<DocumentSectionDef> forType(ConsultationType type) => switch (type) {
        ConsultationType.admissionInterview => admission,
        ConsultationType.treatmentOrders => treatmentOrders,
        ConsultationType.evolution => evolution,
      };
}
